phoenix-core-client/src/main/java/org/apache/phoenix/util/InstanceResolver.java

```diff
@@ -71,14 +70,22 @@ public static <T> T getSingleton(Class<T> clazz, T defaultInstance) {
     public static <T> List get(Class<T> clazz, List<T> defaultInstances) {
         Iterator<T> iterator = ServiceLoader.load(clazz).iterator();
         if (defaultInstances != null) {
-            defaultInstances.addAll(IteratorUtils.toList(iterator));
+            defaultInstances.addAll(iteratorToList(iterator));
         } else {
-            defaultInstances = IteratorUtils.toList(iterator);
+            defaultInstances = iteratorToList(iterator);
         }
 
         return defaultInstances;
     }
 
+    private static <T> List<T> iteratorToList(Iterator<T> iterator) {
```

Comment:
nit: While there is nothing wrong with this, here are a few solutions that do not need a new method:

https://stackoverflow.com/questions/10117026/convert-iterator-to-list

Reply:
I mean the guava or lambd based ones.

Reply:
done